parkhangs-webapp/src/features/parks/parksSlice.js

```diff
@@ -128,12 +153,83 @@ const parksSlice = createSlice({
                 }
             }
         }
+        
+    },
+    extraReducers: {
+        /* 
+            [fetchEventsById.pending]: (state, action) => {
+                if (state.loading === 'idle') {
+                  state.loading = 'pending'
+                  state.currentRequestId = action.meta.requestId
+                }
+            },
+            
+            prepare(events) {
+               return {
+                   payload: {
+                       events //debug
+                   }
+               }
+           }, */
+        
+        
+        [fetchEventsById.fulfilled]:(state, action) =>{
+               //action should return endpoint's call's events
+               //TODO: Currently undefined payload? Action not called?
+               //Or have this action return the events locally to component later?
+               
+                //const {events} = action.payload
+                const { requestId } = action.meta
+
+                console.log("in getEventsByIdSuccess,", action.payload);
+                //console.log("events:" ,events); //broken
+               
+
+                for(let i = 0 ; i< action.payload.length; i ++){
```

Comment:
I'm a bit confused about this, when you get an array of events from a given park, are you adding it to the potential events you've already gotten for another park id?

Reply:
I'm grabbing the array of events which is the action.payload and pushing them to an EventsById array that exists in the store (initial state of park slice). Or I believe that's what I'm doing.

Reply:
sorry my question is about what you're aiming to store - in this case, if there is an array of events already in the store (likely from another park you clicked on), then you'll add the new events to the array, so you'll have 2 parks worth of events stored in parks.eventsById. is that what you intended to have or do you only want 1 parks events in that reducer at a given time?